scripts/tasks.py

```diff
@@ -509,3 +510,242 @@ def timetool_correct(config):
+            else:
+                # Append execution time to the existing dataset
+                existing_execution_times = f['execution_times'][:]
+                f['execution_times'].resize((len(existing_execution_times) + 1,))
+                f['execution_times'][-1] = execution_time
+
+            # Check if 'compression_loss_values' dataset exists, create if not
+            if 'compression_loss_values' not in f:
+                f.create_dataset('compression_loss_values', data=np.array(compression_loss_values))
+            else:
+                # Append compression loss to the existing dataset
+                existing_compression_loss_values = f['compression_loss_values'][:]
+                f['compression_loss_values'].resize((len(existing_compression_loss_values) + 1,))
+                f['compression_loss_values'][-1] = compression_loss
+
+            # Check if 'norm_diff_U_list' dataset exists, create if not
+            if 'norm_diff_U_list' not in f:
+                f.create_dataset('norm_diff_U_list', data=np.array(norm_diff_U_list))
+            else:
+                # Append norm_diff_U_list to the existing dataset
+                existing_norm_diff_U_list = f['norm_diff_U_list'][:]
+                f['norm_diff_U_list'].resize((len(existing_norm_diff_U_list) + 1,))
+                f['norm_diff_U_list'][-1] = np.linalg.norm(diff_U)
+
+            # Check if 'norm_diff_S_list' dataset exists, create if not
+            if 'norm_diff_S_list' not in f:
+                f.create_dataset('norm_diff_S_list', data=np.array(norm_diff_S_list))
+            else:
+                # Append norm_diff_S_list to the existing dataset
+                existing_norm_diff_S_list = f['norm_diff_S_list'][:]
+                f['norm_diff_S_list'].resize((len(existing_norm_diff_S_list) + 1,))
+                f['norm_diff_S_list'][-1] = np.linalg.norm(diff_S)
+
+            # Check if 'norm_diff_mu_list' dataset exists, create if not
+            if 'norm_diff_mu_list' not in f:
+                f.create_dataset('norm_diff_mu_list', data=np.array(norm_diff_mu_list))
+            else:
+                # Append norm_diff_mu_list to the existing dataset
+                existing_norm_diff_mu_list = f['norm_diff_mu_list'][:]
+                f['norm_diff_mu_list'].resize((len(existing_norm_diff_mu_list) + 1,))
+                f['norm_diff_mu_list'][-1] = np.linalg.norm(diff_mu)
+
+            # Check if 'norm_diff_var_list' dataset exists, create if not
+            if 'norm_diff_var_list' not in f:
+                f.create_dataset('norm_diff_var_list', data=np.array(norm_diff_var_list))
+            else:
+                # Append norm_diff_var_list to the existing dataset
+                existing_norm_diff_var_list = f['norm_diff_var_list'][:]
+                f['norm_diff_var_list'].resize((len(existing_norm_diff_var_list) + 1,))
+                f['norm_diff_var_list'][-1] = np.linalg.norm(diff_var)
+
+
+def pipca_online(config):
```

Comment:
Redundancy is high between pipca_online and pipca_offline. Let's try to fix this.

Reply:
I changed the function using this:

```
offline = task.offline

    if offline:
        start_run = run
        num_run = task.max_run
        
    else:
        start_run = run
        num_run = 1
```
In the config file we specify if we want to use the "mode" offline or online.